py/selenium/webdriver/firefox/webdriver.py

```diff
@@ -114,3 +116,25 @@ def firefox_profile(self):
 
     def set_context(self, context):
         self.execute("SET_CONTEXT", {"context": context})
+
+    @contextmanager
+    def using_context(self, context):
+        """Sets the context that Selenium commands are running in using
+        a `with` statement. The state of the context on the server is
+        saved before entering the block, and restored upon exiting it.
+
+        :param context: Context, may be one of the class properties
+            `CONTEXT_CHROME` or `CONTEXT_CONTENT`.
+
+        Usage example::
+
+            with self.selenium.using_context(self.selenium.CONTEXT_CHROME):
+                # chrome scope
+                ... do stuff ...
+        """
+        current_context = self.execute('GET_CONTEXT').pop('value')
```

Comment:
Nit: I would avoid using "current" here because it's misleading once the context changes. The Marionette test I suspect this was based on uses "scope", or you could indicate this is the original or initial context.